src/components/display/fewlines/ShowIdentity/ShowIdentity.tsx

```diff
@@ -60,18 +70,72 @@ export const ShowIdentity: React.FC<ShowIdentityProps> = ({ identity }) => {
         </Link>
       )}
       {!primary && status === "validated" && (
-        <Button variant={ButtonVariant.SECONDARY}>
-          Make this my primary {type}
-        </Button>
+        <>
+          <Button
+            variant={ButtonVariant.SECONDARY}
+            onClick={() => setPrimaryConfirmationBoxOpen(true)}
+          >
+            Make this my primary {type}
+          </Button>
+          {primaryConfirmationBoxOpen && (
+            <ClickAwayListener
+              onClick={() => setPrimaryConfirmationBoxOpen(false)}
+            />
+          )}
+          <ConfirmationBox open={primaryConfirmationBoxOpen}>
+            <p className="confirmation-text">
+              You are about to replace mail@mail.com as your main address
+            </p>
```

Comment:
Maybe make this as a `<ConfirmationText/>`. What do you think?

Reply:
OK I created a new component `<ConfirmationText />` in its own file and exported to stories, `<ShowIdentity />` component and tests 693fcec

Reply:
Do you think you will re-use this component? If so then you can leave it in its own file.

Reply:
I don't think I'll use it outside of a `<ConfirmationBox />`, may I move those 2 components to the same file?

Reply:
Yeah, you can put that back in `ShowIdentity`.

Reply:
I would stand for putting it in the same file as `<ConfirmationBox />` instead. Since I'm using `<ConfirmationText />` also in the stories.
Won't it look weird if I import a component from a page ? What is your opinion?

Reply:
It also feels weird declare something and not use it in the current file, don't you think? This would just create an extra import in `ShowIdentity`.

Reply:
OK fixed d53c644